proposed/2023/Add-dotnet-search.md

```diff
@@ -0,0 +1,94 @@
+# **Add `search` Functionality to DotNet**
+<!-- Replace `Title` with an appropriate title for your design -->
+
+- [Nigusu](https://github.com/Nigusu-Allehu) <!-- GitHub username link -->
+- [GitHub Issue](https://github.com/NuGet/Home/issues/6060) <!-- GitHub Issue link -->
+
+## Summary
+
+<!-- One-paragraph description of the proposal. -->
+The specification seeks to add search functionality to dotnet. The functionality will be added to dotnet as a search command: `dotnet package search`.  
+
+## Motivation
+
+<!-- Why are we doing this? What pain points does this solve? What is the expected outcome? -->
+Customers have expressed a desire for the functionality of the nuget.exe list command to be integrated into the dotnet CLI. Here is a discussion on the request: [NuGet/Home#6060](https://github.com/NuGet/Home/issues/6060). By implementing the search feature, we can effectively address this need, as it will allow users to retrieve and display a list of packages, thereby achieving the primary objective of the list command in a more streamlined manner within the dotnet environment.The [NuGet Server API Search Resource specification requires all packages to be returned when no `q` parameter is provided](https://learn.microsoft.com/en-us/nuget/api/search-query-service-resource#request-parameters). Therefore, this proposal does not add a `list` equivalent to the `dotnet` CLI, as it is effectively redundant.
+
+## Explanation
+
+### Functional explanation
```

Comment:
If the latest version of the package has been deprecated or is vulnerable, does the command output that information? It would be helpful so that customers can make informed decisions in case they would like to install that package.

Reply:
That is a good point raised Kartheek. Currently, no. The reason is that I plan to implement a `--format` option that would allow users to print out the search results in the format they want. That would allow users to be able to use the output in a script or pipeline easily.